docs/index.rst

```diff
@@ -106,24 +106,80 @@ You can use pip and make to quickly set up a development environment::
 
 Please adhere to the following guidelines:
 
-* Lint your code with flake8 and pylint. This can be done easily by executing
-  ``make lint``. The former linter must pass with zero warnings, and warnings
-  from the latter linter should be minimized.
-* Check for errors by running the unit test suite. This can be done easily by
-  executing ``make test``. It must pass with zero errors.
-* Document your code. Documentation can be easily generated by executing ``make
-  docs-html``.
-* Unit tests are *very* highly recommended.
+* All PR’s should follow the predetermined pull request template, explaining the issue, same follows for the Issue if one is to be filed.
+* Maintain Coding Standards
+    * Keep pep8 rules
+    * Follow the same stylistic and logical patterns used in the code
+        * All entity class names and class attributes have to be in the singular format
+        * All required entity attributes have to have `required=True` parameter
+        * It is preferable to use `alpha` data type for default string values for easier debug procedure
+        * In case any workaround is introduced, it is necessary to provide corresponding BZ ID directly into the code docstring
+        * All linting and formatting/style checks would be enforced by Travis-CI and PR would be considered broken until checks are passed successfully.
 * Adhere to typical commit guidelines:
-
-    * Commits should not cause NailGun to break.
+    * Commits should not cause NailGun’s unit test to fail. If it does, it will the responsibility of contributor to fix them.
     * Commits should be small and coherent. One commit should address one issue.
     * Commits should have `good commit messages`_.
     * `Rebasing`_ is encouraged. Rebasing produces a much nicer commit history
       than merging.
-
+* The output of entity operations is a must, either use Satellite or (Foreman+Katello). Outputs should be provided for every new/affected helper or field in the entity. Or, if you have some tests exercising corresponding fields/helpers (for robottelo or other repo) - providing just test results is enough. This should make the review process easier for all reviewers and anyone else interested in the new functionality.
```

Comment:
`The output of entity operations is a must` is somewhat conflicting with `if you have some tests exercising corresponding <SNIP> providing just test results is enough.`

Maybe something like this would be clear? (I'm assuming test results are enough)

>To make the review process easy for all reviewers and anyone else interested in the new functionality please provide some output making use of your changes, this could be in the form of a python interactive shell, test results from Robottelo or Foreman Ansible modules or soemthing else! However you instead to make use of your change will be just fine!